crates/rethnet_evm/src/db/remote.rs

```diff
@@ -0,0 +1,119 @@
+use tokio::runtime::{Builder, Runtime};
+
+use rethnet_eth::remote::{RpcClient, RpcClientError};
+use rethnet_eth::{Address, B256, U256};
+
+use revm::{db::DatabaseRef, AccountInfo, Bytecode};
+
+pub struct RemoteDatabase {
+    client: RpcClient,
+    runtime: Runtime,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum RemoteDatabaseError {
+    #[error("The requested method does not have an implementation")]
+    UnimplementedMethod(),
+
+    #[error("The storage value could not be retrieved")]
+    NetworkError(String),
+
+    #[error("The retrieved value did not match data type exepectations")]
+    UnexpectedResponse {
+        msg: String,
+        request_body: String,
+        expected_type: String,
+        response_text: String,
+    },
+
+    /// Some other error from an underlying dependency
+    #[error(transparent)]
+    OtherError(#[from] std::io::Error),
+}
+
+impl RemoteDatabase {
+    pub fn _new(url: &str) -> Self {
+        Self {
+            client: RpcClient::new(url),
+            runtime: Builder::new_multi_thread()
+                .enable_io()
+                .enable_time()
+                .build()
+                .expect("failed to construct async runtime"),
+        }
+    }
+
+    fn map_err(&self, err: RpcClientError) -> RemoteDatabaseError {
+        match err {
+            RpcClientError::SendError {
+                msg,
+                request_body: _,
```

Comment:
FYI, this code isn't here anymore, because now I'm using  `thiserror`'s `transparent` functionality so that I don't have to re-code any field names here.